tests/unit/test_application.py

```diff
@@ -693,5 +696,203 @@ def test_retry_on_429_status(self):
             self.assertEqual(mock_backoff.call_count, mock_send.call_count)
 
 
+class TestAsyncClient:
+    @pytest.mark.asyncio
+    async def test_total_timeout(self):
+        app = Vespa(url="http://localhost", port=8080)
+        total_timeout = 1  # seconds
+        vespa_async = VespaAsync(app=app, total_timeout=total_timeout)
+
+        # Patch httpx.AsyncClient in the module where VespaAsync is defined
+        with patch(
+            "vespa.application.httpx.AsyncClient", autospec=True
+        ) as MockAsyncClient:
+            # Create an instance of the mock AsyncClient
+            mock_client_instance = MockAsyncClient.return_value
+            # Ensure that 'post' is an AsyncMock
+            mock_client_instance.post = AsyncMock(
+                side_effect=httpx.ReadTimeout("Read timeout")
+            )
+
+            with pytest.raises(tenacity.RetryError) as exc_info:
+                async with vespa_async:
+                    await vespa_async.query(
+                        body={
+                            "yql": "select * from sources * where title contains 'music';"
+                        }
+                    )
+
+            assert isinstance(
+                exc_info.value.last_attempt.exception(), httpx.ReadTimeout
+            )
+
+    @pytest.mark.asyncio
+    async def test_read_timeout(self):
+        app = Vespa(url="http://localhost", port=8080)
+        read_timeout = 1  # seconds
+        vespa_async = VespaAsync(app=app, total_timeout=None, read_timeout=read_timeout)
+
+        with patch(
+            "vespa.application.httpx.AsyncClient", autospec=True
+        ) as MockAsyncClient:
+            mock_client_instance = MockAsyncClient.return_value
+            mock_client_instance.post = AsyncMock(
```

Comment:
Do we have a better way to simulate timeout than mock throwing an exception directly?

Reply:
Hmm, agree that would be nice. I'll try to see if I can find a more relevant way.